bson/marshal.go

```diff
@@ -34,7 +34,7 @@ type Marshaler interface {
 // create custom BSON marshaling behavior for an entire BSON document, implement
 // the Marshaler interface instead.
 type ValueMarshaler interface {
-	MarshalBSONValue() (Type, []byte, error)
+	MarshalBSONValue() (byte, []byte, error)
```

Comment:
Optional: Consider adding named returns to the interface to describe what the values are, since we're removing some type information. We may also want to update the documentation.

```suggestion
	MarshalBSONValue() (typ byte, data []byte, err error)
```